plugins/catalog-react/src/components/UserListPicker/UserListPicker.tsx

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { Fragment, useEffect, useMemo, useState } from 'react';
+import { compact } from 'lodash';
+import { configApiRef, IconComponent, useApi } from '@backstage/core';
+import { UserListFilter, UserListFilterKind } from '../../types';
+import {
+  useEntityListProvider,
+  useOwnUser,
+  useStarredEntities,
+} from '../../hooks';
+import {
+  Card,
+  List,
+  ListItemIcon,
+  ListItemSecondaryAction,
+  ListItemText,
+  makeStyles,
+  MenuItem,
+  Theme,
+  Typography,
+} from '@material-ui/core';
+import SettingsIcon from '@material-ui/icons/Settings';
+import StarIcon from '@material-ui/icons/Star';
+import { reduceEntityFilters } from '../../utils';
+
+const useStyles = makeStyles<Theme>(theme => ({
+  root: {
+    backgroundColor: 'rgba(0, 0, 0, .11)',
+    boxShadow: 'none',
+    margin: theme.spacing(1, 0, 1, 0),
+  },
+  title: {
+    margin: theme.spacing(1, 0, 0, 1),
+    textTransform: 'uppercase',
+    fontSize: 12,
+    fontWeight: 'bold',
+  },
+  listIcon: {
+    minWidth: 30,
+    color: theme.palette.text.primary,
+  },
+  menuItem: {
+    minHeight: theme.spacing(6),
+  },
+  groupWrapper: {
+    margin: theme.spacing(1, 1, 2, 1),
+  },
+}));
+
+export type ButtonGroup = {
+  name: string;
+  items: {
+    id: 'owned' | 'starred' | 'all';
+    label: string;
+    icon?: IconComponent;
+  }[];
+};
+
+function getFilterGroups(orgName: string | undefined): ButtonGroup[] {
+  return [
+    {
+      name: 'Personal',
+      items: [
+        {
+          id: 'owned',
+          label: 'Owned',
+          icon: SettingsIcon,
+        },
+        {
+          id: 'starred',
+          label: 'Starred',
+          icon: StarIcon,
+        },
+      ],
+    },
+    {
+      name: orgName ?? 'Company',
+      items: [
+        {
+          id: 'all',
+          label: 'All',
+        },
+      ],
+    },
+  ];
+}
+
+export const UserListPicker = () => {
+  const classes = useStyles();
+  const configApi = useApi(configApiRef);
+  const orgName = configApi.getOptionalString('organization.name') ?? 'Company';
+  const filterGroups = getFilterGroups(orgName);
+
+  const { value: user } = useOwnUser();
+  const { isStarredEntity } = useStarredEntities();
+
+  // Static filters; used for generating counts of potentially unselected kinds
+  const ownedFilter = useMemo(
+    () => new UserListFilter('owned', user, isStarredEntity),
```

Comment:
Would separate `OwnedEntitiesFilter` and `StarredEntitiesFilter` make sense at this point? Seems like the only thing that requires the current `UserListFilter` is the title handling, which we could probably skip or handle differently?

Reply:
Hmm, the main reason not to imo is that the `CatalogPage` with `initiallySelectedFilter` would be more awkward to handle if it was split. And the title, as you say, which I’d planned to replace with selected-filter chips but not yet in this PR.